### PR TITLE
Raise and reraise exceptions with Stdlib rather than Lwt

### PR DESCRIPTION
**Pull request by MisterDA**
Lwt's documentation reads:

> In most cases, it is better to use `failwith s` from the standard
> library.

and

> Whenever possible, it is recommended to use `raise exn` instead, as
> raise captures a backtrace, while `Lwt.fail` does not. If you call
> `raise exn` in a callback that is expected by Lwt to return a
> promise, Lwt will automatically wrap `exn` in a rejected promise,
> but the backtrace will have been recorded by the OCaml runtime.
>
> For example, `bind`'s second argument is a callback which returns a
> promise. And so it is recommended to use `raise` in the body of that
> callback.
>
> Use `Lwt.fail` only when you specifically want to create a rejected
> promise, to pass to another function, or store in a data structure.

Prefer to capture backtraces to improve debugability.

See also https://github.com/ocsigen/lwt/pull/1008.

### CODE REVIEW

**Review by talex5 — COMMENTED** [review]
Seems reasonable.

File: lib/btrfs_store.ml [hunk 1]
```diff
@@ -151,10 +148,10 @@ let build t ?base ~id fn =
        Btrfs.subvolume_delete result_tmp >>= fun () ->
        Lwt.return r
     )
-  (fun ex ->
```
[the remark this answers is not in the document]
Comment (MisterDA) [reply lib/btrfs_store.ml:154]: I think I saw a mix of `ex` and `exn` and I prefer the latter. I can keep `ex` instead.

I'm not sure why some runs of the CI fail, have I converted one too many call sites?

**Review by MisterDA — COMMENTED** [review]
I think this is good to go.

**Review by talex5 — APPROVED** [review]
Looks good. The `Unix.ENOENT` error from CI is also affecting the main branch, so not due to this PR.

**Comment by shonfeder** [comment]
> @shonfeder I can't merge this because of the new rules preventing merges if the CI doesn't pass.

The system works! :D Tho unfortunately the CI failures here are just https://github.com/ocurrent/obuilder/issues/186.

**Comment by shonfeder** [comment]
Thanks again for this fix :D

**Pull request merged**